diff-hl-show-hunk.el

```diff
@@ -0,0 +1,326 @@
+;;; diff-hl-show-hunk.el --- Integrate popup/posframe and diff-hl-diff-goto-hunk -*- lexical-binding: t -*-
+
+;; Copyright (C) 2020  Free Software Foundation, Inc.
+
+;; Author: Álvaro González <alvarogonzalezsotillo@gmail.com>
+
+;; This file is part of GNU Emacs.
+
+;; GNU Emacs is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; GNU Emacs is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; `diff-hl-show-hunk' shows a posframe/popup with the modified hunk at point.
+;; `diff-hl-show-hunk-function' contains the backend used to show the hunk.  Its
+;; default value is `diff-hl-show-hunk-function-default', which tries to use
+;; `diff-hl-show-hunk-posframe' (GUI), and then `diff-hl-show-hunk-popup'.
+;; Other backends (for example pos-tip or phantom overlays) could be
+;; implemented.
+;;
+;; It uses the same diff algorithm as `diff-hl-flydiff-mode', so it is
+;; highly recomended to enable `diff-hl-flydiff-mode' before use this package.
```

Comment:
Also about this paragraph.

I'd say it's phrased too strongly. I mean, `diff-hl-flydiff-mode` is good and all, but IIUC there won't be any serious cases when you pop up the current hunk and see something unexpected (e.g. a hunk that is different). Will there be?

Because if the original logic in `diff-hl` the hunks that are still highlighted on the fringe (or margin) must all be up-to-date.

Reply:
I was worried about inconsistencies between the diffs showed in the margin and the diffs in the popups.
Maybe It is phrased too strongly (sorry, english is not my best language). I am removing this comment about `diff-hl-flydiff-mode`.